homeassistant/components/rainbird/__init__.py

```diff
@@ -41,10 +53,13 @@ def setup(hass, config):
 
     _LOGGER.debug("Rain Bird Controller set to: %s", server)
 
-    initial_status = controller.command("ModelAndVersion")
-    if initial_status and initial_status["type"] != "ModelAndVersionResponse":
-        _LOGGER.error("Error getting state. Possible configuration issues")
-        return False
+    for platform in [switch.DOMAIN, sensor.DOMAIN, binary_sensor.DOMAIN]:
+        discovery.load_platform(
+            hass,
+            platform,
+            DOMAIN,
+            discovered={RAINBIRD_CONTROLLER: controller},
```

Comment:
Only serializable info can be sent. Store the controller in hass.data instead like before.

Pass an empty dict as discovery_info.

Reply:
done